### DISCUSSION TITLE
Are `prisma.$transactions` wrapped inside `BEGIN` and `COMMIT` for postgres?

### DISCUSSION THREAD

**Discussion by aarontravass** [prisma/prisma#23780] [Q&A] [answered]
### Question

Given this statement in the [docs](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#transaction-api)

```log
The $transaction([]) API is generic solution to independent writes that allows you to run multiple operations as a single, atomic operation - if any operation fails, Prisma Client rolls back the entire transaction.
```

are transactions for postgres wrapped inside of a `BEGIN`/`COMMIT`? If not, then how are they sent to the database?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @aarontravass 👋 

Thank you for raising this question.

Prisma transactions are wrapped inside a `BEGIN...COMMIT/ROLLBACK` statement. You can enable the query [`log`](https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/logging#log-to-stdout) to see the statements when you execute your Prisma query. 

```js
const prisma = new PrismaClient({log: ["info", "query"]})

 const id = 9 // User to be deleted

const deletePosts = prisma.post.deleteMany({
  where: {
    userId: id,
  },
})

const deleteMessages = prisma.privateMessage.deleteMany({
  where: {
    userId: id,
  },
})

const deleteUser = prisma.user.delete({
  where: {
    id: id,
  },
})

const res = await prisma.$transaction([deletePosts, deleteMessages, deleteUser]) // Operations succeed or fail together
  console.dir( res, { depth: Infinity })
```

The output should look like:

```js
prisma:info Starting a postgresql pool with 17 connections.
prisma:query BEGIN
prisma:query DELETE FROM "public"."Post" WHERE "public"."Post"."userId" = $1
prisma:query DELETE FROM "public"."PrivateMessage" WHERE "public"."PrivateMessage"."userId" = $1
prisma:query DELETE FROM "public"."User" WHERE ("public"."User"."id" = $1 AND 1=1) RETURNING "public"."User"."id"
prisma:query ROLLBACK
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by aarontravass** [reply]
Thanks
